app/routers/quizzes.py

```diff
@@ -92,6 +93,35 @@ async def get_quiz(quiz_id: str):
     quiz_collection = client.quiz.quizzes
     if (quiz := quiz_collection.find_one({"_id": quiz_id})) is not None:
         update_quiz_for_backwards_compatibility(quiz_collection, quiz_id, quiz)
+
+        if quiz["metadata"]["quiz_type"] == QuizType.omr.value:
+            print("true", QuizType.omr)
+            for question_set_index, question_set in enumerate(quiz["question_sets"]):
+                options_count_aggregate = client.quiz.questions.aggregate(
+                    [
+                        {"$match": {"question_set_id": question_set["_id"]}},
+                        {"$sort": {"_id": 1}},
+                        {"$skip": settings.subset_size},
+                        {"$project": {"number_of_options": {"$size": "$options"}}},
+                    ]
+                )
```

Comment:
You're running this aggregation query 4 times (for 4 question sets). Everytime mongo has to parse through ALL the questions in the collection of the questions 4 times. 
You can club this into one aggregation query.
```
question_set_ids = [ q_set["_id"] for q_set in quiz["question_sets"] ]
options_count_aggregate = client.quiz.questions.aggregate(
    [
        {"$match": {"question_set_id": {
            "$in": question_set_ids
        }}},
        {"$sort": {"_id": 1}},
        {"$skip": settings.subset_size},
        {"$project": {"number_of_options": {"$size": "$options"}}},
    ]
)
```

Now you have option_counts for all the questions belonging to all the question_sets in one array.
You should also project `question_set_id` in this case so you can parse and store it

Reply:
Also add a comment on top of this no, what are you trying to do here.
Took some time to understand what information will `options_count_aggregate` will store